src/client/session/mod.rs

```diff
@@ -118,12 +118,14 @@ pub(crate) struct Transaction {
     pub(crate) state: TransactionState,
     pub(crate) options: Option<TransactionOptions>,
     pub(crate) pinned_mongos: Option<SelectionCriteria>,
+    pub(crate) recovery_token: Option<Document>,
 }
 
 impl Transaction {
     pub(crate) fn start(&mut self, options: Option<TransactionOptions>) {
         self.state = TransactionState::Starting;
         self.options = options;
+        self.recovery_token = None;
```

Comment:
Is this assignment necessary? Similarly to `pinned_mongos` I believe this field should be `None` by default upon the first transaction and reset to `None` after previous transactions have finished.

Reply:
Yeah. From [the spec](https://github.com/mongodb/specifications/blob/master/source/transactions/transactions.rst#recoverytoken-field):
> Drivers MUST clear a session's cached `recoveryToken` when transitioning to the "no transaction" or "starting transaction" state.

So, I'm clearing the `recoveryToken` on `Transaction::start` and `Transaction::reset`